File: src/main/java/entry/JsonStructBuilderTest.java
package entry;

import org.junit.Test;

import static org.junit.Assert.*;

public class JsonStructBuilderTest {
    JsonStructBuilder builder = new JsonStructBuilder("Gen");

    @Test
    public void set_title() {
    }

    @Test
    public void gen() {

        String json6 = "{\n" +
                "  \"aa\": 1,\n" +
                "\t\"bb\": [{\"c\": 1}],\n" +
                "\t\"cc\": {\"d\": 1},\n" +
                "\t\"dd\":[],\n" +
                "\t\"ee\": [1],\n" +
                "\t\"ff_hello\": [{\"f1\": {\"f2\": [1]}}]\n" +
                "}";
        String out = this.builder.gen(json6);
        System.out.println(out);


    }

    @Test
    public void makeField() {
    }

    @Test
    public void build() {
    }
}

File: src/main/kotlin/entry/JsonStructBuilderTest.kt
package entry

import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertSame

class JsonStructBuilderTest {

    @Test
    fun gen() {
        var builder = JsonStructBuilder("T")
        var result = builder.gen("[{}]")
        assertEquals(
            result,
            """type T struct {
}"""
        )
        builder = JsonStructBuilder("T")
        result = builder.gen("""{}""")
        assertEquals(
            result,
            """type T struct {
}"""
        )

        builder = JsonStructBuilder("T")
        builder.setConfig("json:\"%s\" bson:\"%s\"", false)
        result = builder.gen("""{"a": 1, "A": "b"}""")
        println(result)
        assertEquals(result, """type T struct {
    A    int    `json:"a" bson:"a"`
    A    string    `json:"A" bson:"A"`
}""")

    }
}